src/components/HomeUpload/HomeUpload.js

```diff
@@ -11,8 +11,33 @@ import AppContext from "../../AppContext";
 export default function HomeUpload() {
   const [files, setFiles] = useState([]);
   const { apiUrl } = useContext(AppContext);
+  const [directoryMode, setDirectoryMode] = useState(false);
+
+  useEffect(() => {
+    if (directoryMode) {
+      inputRef.current.setAttribute("webkitdirectory", "true");
+    } else {
+      inputRef.current.removeAttribute("webkitdirectory");
+    }
+  }, [directoryMode]);
+
+  const getFilePath = (file) => {
```

Comment:
If this function gets called on a file that's not uploaded through a dir upload, I believe it fails.
Could we add sanity checks here or make that apparent either through a comment or variable name?

Reply:
It's not going to fail, we are actually calling it for every file, regardless of whether it was directory or files upload - https://github.com/NebulousLabs/skynet-webportal/pull/60/files#diff-4f89bf10d6d343f05c6bdb8c1edb1135R72

Reply:
hm I don't understand how that works because of the `slice(1)` in case of plain files...

Reply:
nevermind! :)